cpp/unreal_plugins/UrdfBot/Source/UrdfBot/UrdfBotPawn.cpp

```diff
@@ -47,9 +48,39 @@ AUrdfBotPawn::AUrdfBotPawn(const FObjectInitializer& object_initializer) : APawn
 void AUrdfBotPawn::SetupPlayerInputComponent(class UInputComponent* input_component)
 {
     Super::SetupPlayerInputComponent(input_component);
+
+    int axis_binding_index = 0;
+    UPlayerInput* player_input = GetWorld()->GetFirstPlayerController()->PlayerInput;
+    std::vector<YAML::Node> keyboard_controls_ = Config::get<std::vector<YAML::Node>>("URDFBOT.URDFBOT_PAWN.ROBOT_COMPONENT.KEYBOARD_CONTROL");
+    for (auto& control : keyboard_controls_) {
+        InputAxisBinding input_axis_binding;
+        input_axis_binding.axis_name_ = FName(GetName() + "_" + FString::FromInt(axis_binding_index));
+
+        input_axis_binding.key_ = control["KEY"].as<std::string>();
+        std::string type = control["TYPE"].as<std::string>();
+        if (type == "set") {
+            input_axis_binding.type_ = AxisBindingType::Set;
+        } else if (type == "add") {
+            input_axis_binding.type_ = AxisBindingType::Add;
+        } else {
+            ASSERT(false);
+        }
+        input_axis_binding.component_names_ = control["COMPONENT"].as<std::vector<std::string>>();
+        input_axis_binding.values_ = control["VALUE"].as<std::vector<float>>();
+        ASSERT(input_axis_binding.component_names_.size() == input_axis_binding.values_.size());
+
+        player_input->AddAxisMapping(FInputAxisKeyMapping(input_axis_binding.axis_name_, FKey(FName(input_axis_binding.key_.c_str())), 1));
+        input_component->BindAxis(input_axis_binding.axis_name_);
+
+        robot_component_->input_axis_bindings_.push_back(input_axis_binding);
+
+        axis_binding_index++;
```

Comment:
- No need for this index. The axis name could just be: "MyUrdfRobotPawn::W", "MyUrdfRobotPawn::A", etc, instead of "MyUrdfRobotPawn_0", "MyUrdfRobotPawn_1", etc.

Reply:
fixed